### ISSUE TITLE
"Oculus user entitlement check failed" on Beat Saber after rebooting my PC

### ISSUE THREAD

**Issue by CodyMarkix** [LibreVR/Revive#1860]
**Do not report issues when using the Virtual Desktop or ALVR drivers for SteamVR, this is not supported by Revive.**

**Describe the bug**
When installing Beat Saber, running via Revive works fine, even after downgrading and modding. After a PC reboot, attempting to launch Beat Saber either via the Steam library or Revive Dashboard gets Beat Saber to the epilepsy warning, then crashes with an entitlement check fail.

**To Reproduce**
1. Open the Oculus app
2. Start SteamVR
3. Open Beat Saber via the Revive dashboard
4. See Beat Saber crash after the epilepsy warning
5. Cry because you've been troubleshooting for 2 days straight

**Expected behavior**
Beat Saber opens up and authenticates with Oculus like intended

**Environment (please complete the following information):**
 - App: Beat Saber 1.29.0
 - OS: Windows 11 23H2
 - Headset: Oculus Quest 2 (ALVR)

**Versions (please complete the following information):**
 - Revive: 3.0.2
 - SteamVR: 2.6.2

**Additional context**
While I'm running an older (and modded) version of Beat Saber downgraded via OculusDowngrader, this also happens on the latest vanilla version. (1.37.0)

Proof of ownership:
![BZ9yupb](https://github.com/LibreVR/Revive/assets/45365053/35e09b72-d78b-4bc9-852c-7ee54ca62af2)

**Comment by CrossVR** [comment]
Are the Oculus VR Runtime Service and the Oculus VR Library Service running in the background like normal? Some people in the past disabled those services and then complained their game wouldn't start anymore due to entitlement errors.

Since you have an Oculus headset, please try running the game normally first before using SteamVR, I assume you're running on SteamVR to use chat overlays? Also, ALVR drivers aren't really supported by Revive, but this seems like an unrelated issue.

**Comment by CodyMarkix** [comment]
I did try using Oculus Link and the game launched as intended. And the VR Library service never starts as far as I can tell, so I'm gonna try to play with it enabled later during the day (currently traveling somewhere, thankfully with my quest 2) and update this comment.

**Comment by CodyMarkix** [comment]
@CrossVR attempting to start the VR Library service gives this error
![image](https://github.com/LibreVR/Revive/assets/45365053/e0f8f16d-d6ac-412a-8bd7-835fcc3fcf71)

**Comment by CodyMarkix** [comment]
Also from what I've read, the library service should be automatically started by the OVR server, which it doesn't. (But perhaps that's irrelevant in the case of Revive?)

**Comment by CodyMarkix** [comment]
![image](https://github.com/LibreVR/Revive/assets/45365053/02bfd7f7-b4e9-4639-8098-5767824a4936)

**Comment by CrossVR** [comment]
I was more concerned with it being purposefully disabled. The fix I wanted to try was to have you start the game through the Oculus store app before attempting to start it through Revive again as this refreshes the entitlement.

**Comment by CodyMarkix** [comment]
No, I never purposefully disabled the service (truth be told I never noticed its existence until now haha). I'm also trying to get zerotier working with oculus link to refresh the entitlement, will update you on that later.
